lib/git/lib.rb

```diff
@@ -457,14 +460,16 @@ def global_config_get(name)
     end
 
     def config_list
-      build_list = lambda do |path|
-        parse_config_list command_lines('config', ['--list'])
-      end
+      @@config_semaphore.synchronize do
+        build_list = lambda do |path|
+          parse_config_list command_lines('config', ['--list'])
+        end
 
-      if @git_dir
-        Dir.chdir(@git_dir, &build_list)
-      else
-        build_list.call
+        if @git_dir
+          Dir.chdir(@git_dir, &build_list)
```

Comment:
Likewise, not thread-safe.

Reply:
Yeah, I thought I took care of these, might just be a merge issue. I'm offline for a week now, but will take a look.

Reply:
@perlun and @taquitos isn't the purpose of the semaphore to make these thread safe?